agent/structs/structs.go

```diff
@@ -2211,8 +2211,8 @@ type PeeredServiceName struct {
 }
 
 type ServiceName struct {
-	Name string
-	acl.EnterpriseMeta
+	Name               string
+	acl.EnterpriseMeta `mapstructure:",squash"`
```

Comment:
Noticed in testing that `mapstructure` does not handle embedded structs without this tag